File: easy/kids_with_greater_number_of_candies.js
var max_of_array = function(arr){
    let max = 0
    for(let i = 0;i<arr.length;i++){
        if(arr[i]>max){
            max = arr[i]
        }
    }
    return max
}
var kidsWithCandies = function(candies, extraCandies) {
    let max = max_of_array(candies)
    let res = []
    for(let i = 0;i<candies.length;i++){
        if(max<=candies[i]+extraCandies){
                res.push(true)  
        }
        else{
                res.push(false)
        }
    }
    return res
};

File: easy/kidsWithGreaterNumberOfCandies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        List<Boolean> r = new ArrayList<>();
        for(int i:candies) max = Math.max(i,max);
        for(int i=0;i<candies.length;i++){
            if(candies[i]+extraCandies>=max){
                r.add(true);
            }
            else{
                r.add(false);
            }
        }
        return r;
    }
    
}
